apollo-federation/src/sources/connect/models.rs

```diff
@@ -365,29 +365,20 @@ fn determine_entity_resolver(
 fn extract_header_references(
     variable_references: HashSet<VariableReference<Namespace>>,
 ) -> HashSet<String> {
-    let headers: HashSet<String> = variable_references
+    variable_references
         .iter()
-        .filter_map(|var_ref| {
+        .flat_map(|var_ref| {
```

Comment:
```rust
.flat_map(|var_ref| var_ref
  .selection.get("headers")
  .map(|headers_subselection| 
    headers_subselection.keys()
      .cloned()
      .collect()
  ).unwrap_or_default()
)
```

Reply:
How would you include checking that the `Namespace` is either `$request` or `$response`?

Reply:
Ah, I see what you mean: https://github.com/apollographql/router/pull/7419/commits/d7e8d5e70b57f1d94ea5d117506c2cc4b235b031